packages/marshal/src/types.js

```diff
@@ -2,70 +2,324 @@
 /// <reference path="extra-types.d.ts" />
 
 /**
- * @typedef { "bigint" | "boolean" | "null" | "number" | "string" | "symbol" | "undefined" | "copyArray" | "copyRecord" | "copyError" | "promise" | "presence" } PassStyle
- * TODO "presence" above should indirect through REMOTE_STYLE to prepare
- * for changing it to "remotable"
+ * @typedef {
+ *   "undefined" | "null" |
+ *   "boolean" | "number" | "bigint" | "string" | "symbol" |
```

Comment:
Would symbols be unique per-host, or would two symbols constructed from the same string on two hosts be treated as the same symbol?

Reply:
Same symbol. That's why it maps only to JS registered symbols and well-known symbol, but excludes JS unregistered symbols. Likewise, why it maps to lisp/scheme interned string symbols but excludes gensyms. The only unforgeable identities are the Remotables.

Reply:
Ahh, so is “passable” vs “remotable” the same as “records” and “interfaces”, as I’ve heard you call them before?

Reply:
Yes, `Remotable` is the analog of the traditional "interface".

But `CopyRecord` is the analog of the traditional "record".

`Passable` includes these and all other things that can be passed as arguments in messages. `Passable` is their common supertype.

Reply:
This is really good news re: symbols, and that they're `eq?` too I think isn't totally required from a protocol perspective, but is a big win in terms of shared semantics!

Reply:
Many languages, even those in the cohort nearest to Agoric/Goblins, will not have an analog for symbols.

Reply:
@kriskowal @cwebber have we resolved the symbol vs string question?

Reply:
I think the jury might still be out. On the one hand, assuming that registered symbols are garbage collected, I think JavaScript and Racket can at least coherently mirror each other’s types. With Python and many other languages, the abstraction would leak: since there’s no language feature, the notion of symbols would be hosted by the CapTP library.

There may be a middle ground where Racket would recognize strings-on-the-wire as symbols-in-memory by their position, like field and method names. We have yet to settle on a representation for records with stringly-named-fields: my Syrup is abusing dictionaries for this purpose, which I believe is across intended purposes.

One thing that helps clear the log jam is that I believe we have loose consensus that CapTP will not impose its own idioms on the expression of API’s, so a Racket schema might favor symbols for method names, where the JavaScript equivalent would strictly use strings, just as the Racket would favor function application where JavaScript would favor method invocation. Speaking across languages would be awkward but possible.

Reply:
Let’s proceed with the hypothesis that symbols are in, so that we can gather evidence about the developer experience with them.